python/tskit/tables.py

```diff
@@ -2850,6 +2856,10 @@ def assert_equals(
                 f" differs: self={self.sequence_length} other={other.sequence_length}"
             )
 
+        # NOTE: Do we need to anything "extra" to account for `ignore_tables`?
```

Comment:
I think this is fine - we don't keep any particular state to say that the tables were created by "ignore_tables" so failing on the number of edges being different is as good as any.